log4j-api/src/main/java/org/apache/logging/log4j/simple/SimpleLoggerContext.java

```diff
@@ -100,7 +99,6 @@ public ExtendedLogger getLogger(final String name, final MessageFactory messageF
         // Note: This is the only method where we add entries to the 'loggerRegistry' ivar.
         final ExtendedLogger extendedLogger = loggerRegistry.getLogger(name, messageFactory);
         if (extendedLogger != null) {
-            AbstractLogger.checkMessageFactory(extendedLogger, messageFactory);
```

Comment:
After the changes in [LOG4J2-1180](https://issues.apache.org/jira/browse/LOG4J2-1180) and [LOG4J2-1318](https://issues.apache.org/jira/browse/LOG4J2-1318) the `checkMessageFactory` method is mostly useless, but there is a corner case.

If I have a **stateful** message factory `MyMessageFactory` (e.g. it creates map messages with a predefined set of keys), then we still need a warning if we call:

```java
Logger logger1 = LogManager.getLogger("foo", new MyMessageFactory("bar", "baz"));
Logger logger2 = LogManager.getLogger("foo", new MyMessageFactory("bar", "car"));
```

The `LoggerRegistry` only uses the **class name** of `MyMessageFactory` as key, so we'll end up with `logger2 == logger1`, even if the user wanted two different loggers.

Reply:
Right. Implemented `MessageFactory`-namespaced logger registry in be6c229307fb907b1c70fbb421c3a8683907452e. Mind reviewing it, please?